src/server/services/wms/qgswmsparameters.h

```diff
@@ -0,0 +1,240 @@
+/***************************************************************************
+                              qgswmsparameters.h
+                              ------------------
+  begin                : March 17, 2017
+  copyright            : (C) 2017 by Paul Blottiere
+  email                : paul dot blottiere at oslandia dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSWMSPARAMETERS_H
+#define QGSWMSPARAMETERS_H
+
+#include <QMap>
+#include <QObject>
+#include <QMetaEnum>
+#include <QColor>
+#include "qgsrectangle.h"
+#include "qgsserverrequest.h"
+
+namespace QgsWms
+{
+  struct QgsWmsParametersLayer
+  {
+    QString mNickname; // name, id or short name
+    int mOpacity = -1;
+    QStringList mFilter; // list of filter
+    QStringList mSelection; // list of string fid
+    QString mStyle;
+  };
+
+  struct QgsWmsParametersHighlightLayer
+  {
+    QString mName;
+    QString mWkt;
+    QString mSld;
+    QString mLabel;
+    QColor mColor;
+    int mSize = 0;
+    int mWeight = 0;
+    QString mFont;
+    float mBufferSize = 0;
+    QColor mBufferColor;
+  };
+
+  class QgsWmsParameters : public QObject
+  {
+      Q_OBJECT
```

Comment:
Why is this a QObject? I can't see any signals/slots/other QObject features here.